### ISSUE TITLE
Doesn't handle "Preserving and Restoring State" properly on macOS.

### ISSUE THREAD

**Issue by luceat-lux-vestra** [jwilm/alacritty#669]
I don't know it's details about this and how to deal with it in code level.

But I want to `alacritty` to remember last window size and position and would be there same size and position when I reopen it.

Please let me know something about it.

**Comment by brycefisher** [comment]
@luceat-lux-vestra yes, this is totally doable to get/set position using the glutin window APIs:

 * https://docs.rs/glutin/0.9.2/glutin/struct.GlWindow.html#method.get_position
 * https://docs.rs/glutin/0.9.2/glutin/struct.GlWindow.html#method.set_position

I don't know if it makes sense to remember the window **size** given we already specify height and width in lines and columns in the config file. There's nothing that governs the window position, so there's no design conflict there.

@jwilm I've started working on branch to add support for remembering window position. I'm not sure if I've hit on the best strategy and wanted run by the high level plan before much further.

## Implementation Plan

 * Add a new top level config `remember_position` (name up for debate)
 * When the window closes write the x and y position of the window in a file adjacent to the config file
 * At new window creation time, if `remember_position` is set to true, check all those locations for a remembered position file. If no remembered position, don't call `window.set_position`

Here's the file locations I considered to match up with how config files are discovered. (I don't love how complex this is, but I assume consistency is more important):

 1. `$XDG_CONFIG/alacritty/.remembered_position.yml`
 2. `$XDG_CONFIG/.alacritty.remembered_position.yml`
 3. `$HOME/.config/alacritty/.remembered_position.yml`
 4. `$HOME/.remembered_position.yml`

## Problems

 * I'm not sure to avoid constantly polling the `GLWindow` object for its position, which could performance regressions 😭 
 * deciding where to write the file to disk gets a bit hairy
 * Testing could be tricky
 * there's so many paths! (That could is tricky to refactor but feels non-DRY to just copy/tweak)

## My Branch
https://github.com/jwilm/alacritty/pull/765

**Comment by chrisduerr** [comment]
There's some additional information in https://github.com/jwilm/alacritty/issues/669.

I've opened a winit issue for tracking a potential upstream implementation: https://github.com/rust-windowing/winit/issues/941
